### PR TITLE
[DPE-2546] Split stanza create and stanza check

### PR DESCRIPTION
**Pull request by dragomirp**
## Issue
Pgbackrest stanza check requires  TLS servers to run if enabled, but they won't start until the hook is released and peer data is updated.

## Solution
Run only stanza create in the s3 relation hooks and trigger the stanza check via peer relation change

### CODE REVIEW

File: src/backups.py [hunk 1]
```diff
@@ -299,12 +299,18 @@ def _initialise_stanza(self) -> None:
             self.charm.unit.status = BlockedStatus(FAILED_TO_INITIALIZE_STANZA_ERROR_MESSAGE)
             return
 
+        self.start_stop_pgbackrest_service()
+
         # Store the stanza name to be used in configurations updates.
-        self.charm.app_peer_data.update({"stanza": self.stanza_name})
+        self.charm.app_peer_data.update({"stanza": self.stanza_name, "init-pgbackrest": "True"})
```
Comment (dragomirp) [inline src/backups.py:305]: An extra flag just in case.

File: src/backups.py [hunk 2]
```diff
@@ -317,13 +323,14 @@ def _initialise_stanza(self) -> None:
...
             self.charm.app_peer_data.update({"stanza": ""})
+            self.charm.app_peer_data.pop("init-pgbackrest", None)
             self.charm.update_config()
 
             logger.exception(e)
             self.charm.unit.status = BlockedStatus(FAILED_TO_INITIALIZE_STANZA_ERROR_MESSAGE)
             return
 
-        return
+        self.charm.app_peer_data.pop("init-pgbackrest", None)
```
Comment (dragomirp) [inline src/backups.py:333]: Clear the flag so we stop checking the stanza.

File: src/charm.py [hunk 3]
```diff
@@ -474,7 +477,7 @@ def _on_peer_relation_changed(self, event: HookEvent) -> None:
             event.defer()
             return
         else:
-            self.unit_peer_data.pop("start-tls.server", None)
+            self.unit_peer_data.pop("start-tls-server", None)
```
Comment (taurus-forever) [inline src/charm.py:480]: This is an interesting catch... 

Q: why do we set/remove it if we do not check it?
```
> ripgrep start-tls.server
src/charm.py
470:            self.unit_peer_data.update({"start-tls-server": "True"})
477:            self.unit_peer_data.pop("start-tls.server", None)
```

**Review by taurus-forever — APPROVED** [review]
LGTM, the idea to seaprat stanza init and check is a proper approach, IMHO.
Thank you!

File: src/charm.py (hunk 3 above)
Comment (dragomirp) [reply src/charm.py:480]: I'm unsure, so I left it be for the time being.

**Comment by cjohnston1158** [comment]
This change resolved the issue for me.

File: src/charm.py [hunk 4]
```diff
@@ -464,6 +464,9 @@ def _on_peer_relation_changed(self, event: HookEvent) -> None:
 
         self.postgresql_client_relation.update_read_only_endpoint()
 
+        if self.unit.is_leader() and "init-pgbackrest" in self.app_peer_data:
```
Comment (delgod) [inline src/charm.py:467]: Should we move the condition inside the `check_stanza` function to not spread the `init-pgbackrest` logic between many different functions?

File: src/backups.py [hunk 5]
```diff
@@ -299,12 +299,18 @@ def _initialise_stanza(self) -> None:
...
         # Store the stanza name to be used in configurations updates.
-        self.charm.app_peer_data.update({"stanza": self.stanza_name})
+        self.charm.app_peer_data.update({"stanza": self.stanza_name, "init-pgbackrest": "True"})
 
+    def check_stanza(self) -> None:
+        """Runs the pgbackrest stanza validation."""
         # Update the configuration to use pgBackRest as the archiving mechanism.
         self.charm.update_config()
 
+        self.charm.unit.status = MaintenanceStatus("checking stanza")
```
Comment (delgod) [inline src/backups.py:312]: Do we really need this status? I think it is ok just put "Checking stanza" into logs.

**Review by delgod — APPROVED** [review]

File: src/backups.py (hunk 5 above)
Comment (dragomirp) [reply src/backups.py:312]: We don't need it, but I think it makes things clearer to the user.

**Review by marceloneppel — APPROVED** [review]

**Pull request merged**